### PR TITLE
Allow reg of teacher users, class shows realnames

### LINKED ISSUE #204: classroom page not showing student names
Teacher now sees username (username) instead of username (First Last).

### LINKED ISSUE #209: Only teacher users may become classroom teachers
The public version seems to now be broken on adding a second teacher to a classroom., always giving the error message in the title. This might be related to issue 204.

### PR DESCRIPTION
**Pull request by hiliang-cmu**
Classroom progress panels and CSV columns show teamname and member name(s), allowing for multi-member teams. 

Panel title:
`team1 (alice smith, bob smith)`

CSV: 

Teamname|Member Name(s)
-|-
`team1`|alice smith, bob smith

* Resolves #209 
* Resolves #204 
* Revert eb31ad4 "Remove/hide classroom e-mail invite from UI"
* Partial revert a1aa23f "Additional adjustments to classroom per #181 issues"

### CODE REVIEW

File: picoCTF-web/web/classroom.html [hunk 1]
```diff
@@ -181,7 +179,7 @@ <h4>Exporting Data</h4>
             </div>
           </div>
           <% _.each(_.sortBy(teams, "team_name"), function(team) {
-            members = team.members.map(function(member){ return member.username }).join(", ") %>
+            members = team.members.map(function(member){ return member.firstname + " " + member.lastname }).join(", ") %>
```
Comment (maverickwoo) [inline picoCTF-web/web/classroom.html]: This (and in `classroom.coffee` below) is where we should introduce a function that *uniformly* returns a human-friendly string to identify a member regardless of whether we collect names or not. We know a member has a `username` and optionally a `firstname` and a `lastname`. The function would return `"({{firstname}} {{lastname}} ({{username}})"` if we have the name, or `"{{username}}"` if we don't. That way, we won't need to change any code until we need to deal with languages that put the last name in the front.

Case 1: "John Doe (jdoe)"
Case 1 missing first name: "Doe (jdoe)"
Case 1 missing last name: "John (jdoe)"
Case 2: "jdoe"

Comment (hiliang-cmu) [reply picoCTF-web/web/classroom.html]: I think we'd only have case 1 and 2, not missing only 1b) fname or 1c) lname. Those are required fields currently (our 2018 competition replaced them with `-`)

**Comment by hiliang-cmu** [comment]
Rebased and addresses concern re: uniform name output as per PR discussion

File: picoCTF-web/web/classroom.html (hunk 1 above)
Comment (hiliang-cmu) [reply picoCTF-web/web/classroom.html]: Addressed by d9a437e

**Comment by dmartin** [comment]
Tested it out, the only potential issue I noticed was that you can only join the classroom with the username of the original teacher who created it.

So if you're the second teacher to join a classroom, the message "You can also provide **your** username and the name of the class to your students" isn't quite valid.

**Comment by hiliang-cmu** [comment]
Changed the text labels in df99bf5 to clarify there is only 1 owner, not all teachers. It's not explicit that the owner is a teacher but I don't think that's a huge problem since only teachers can create classrooms.

**Pull request merged**